lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -286,6 +251,59 @@ public void OnNext(ICompletedTask completedTask)
             }
         }
 
+        public void OnNext(IFailedEvaluator value)
+        {
+            Logger.Log(Level.Info, "An evaluator failed, checking if it failed before context and service was submitted");
+            int currFailedEvaluators = Interlocked.Increment(ref _currentFailedEvaluators);
+
+            if (value.FailedTask != null && value.FailedTask.IsPresent())
+            {
+                Exceptions.Throw(new Exception("An evaluator failed during task, cannot continue IMRU task"), Logger);
+                Logger.Log(Level.Error, "An evaluator failed during task, cannot continue IMRU task");     
+            }
+
+            if (value.FailedContexts != null && value.FailedContexts.Count != 0)
```

Comment:
If there is a task, there is also a context. Hence, you could skip the check for task presence.